### PR TITLE
Added variant for Waveshare ESP32-S3-Touch-LCD-1.69 & ESP32-S3-LCD-1.69 board

### PR DESCRIPTION
**Pull request by Y1hsiaochunnn**
## Description of Change
Adding variant for Waveshare ESP32-S3-Touch-LCD-1.69 & ESP32-S3-LCD-1.69 board containing MCU, LCD, RTC, and IMU.

## Tests scenarios
Untested, pin definition is based on information in Waveshare wiki.

## Related links

Wiki For [ESP32-S3-Touch-LCD-1.69](https://www.waveshare.com/wiki/ESP32-S3-Touch-LCD-1.69)
Wiki For [ESP32-S3-LCD-1.69](https://www.waveshare.com/wiki/ESP32-S3-LCD-1.69)

Pending USB VID PR: [espressif/usb-pids/pull/169](https://github.com/espressif/usb-pids/pull/169)

### CODE REVIEW

**Comment by P-R-O-C-H-Y** [comment]
@Y1hsiaochunnn CI is failing as the ESP32-S3-LCD-1.69 is missing SPI default pins. Can you please add those. Thanks

**Comment by Y1hsiaochunnn** [comment]
my bad, I'm a little confused about the configuration of the two boards,Now that I've reworked it, it should work just fine

File: boards.txt
Comment (P-R-O-C-H-Y) [inline boards.txt:38067]:
```suggestion
waveshare_esp32_s3_touch_lcd_169.build.flash_size=16MB
```

File: boards.txt
Comment (P-R-O-C-H-Y) [inline boards.txt:38084]:
```suggestion
waveshare_esp32_s3_touch_lcd_169.menu.PSRAM.enabled.build.psram_type=opi
```

File: boards.txt
Comment (P-R-O-C-H-Y) [inline boards.txt:38098]:
```suggestion
```

File: boards.txt
Comment (P-R-O-C-H-Y) [inline boards.txt:38137]:
```suggestion
waveshare_esp32_s3_touch_lcd_169.menu.PartitionScheme.app3M_fat9M_16MB=16M Flash (3MB APP/9.9MB FATFS)
waveshare_esp32_s3_touch_lcd_169.menu.PartitionScheme.default=Default 4MB with spiffs (1.2MB APP/1.5MB SPIFFS)
```

File: boards.txt
Comment (P-R-O-C-H-Y) [inline boards.txt:38165]:
```suggestion
waveshare_esp32_s3_touch_lcd_169.menu.PartitionScheme.rainmaker_8MB=RainMaker 8MB
waveshare_esp32_s3_touch_lcd_169.menu.PartitionScheme.rainmaker_8MB.build.partitions=rainmaker_8MB
waveshare_esp32_s3_touch_lcd_169.menu.PartitionScheme.rainmaker_8MB.upload.maximum_size=4116480
waveshare_esp32_s3_touch_lcd_169.menu.PartitionScheme.fatflash=16M Flash (2MB APP/12.5MB FATFS)
waveshare_esp32_s3_touch_lcd_169.menu.PartitionScheme.fatflash.build.partitions=ffat
waveshare_esp32_s3_touch_lcd_169.menu.PartitionScheme.fatflash.upload.maximum_size=2097152
waveshare_esp32_s3_touch_lcd_169.menu.PartitionScheme.app3M_fat9M_16MB=16M Flash (3MB APP/9.9MB FATFS)
waveshare_esp32_s3_touch_lcd_169.menu.PartitionScheme.app3M_fat9M_16MB.build.partitions=app3M_fat9M_16MB
waveshare_esp32_s3_touch_lcd_169.menu.PartitionScheme.app3M_fat9M_16MB.upload.maximum_size=3145728

```

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
@Y1hsiaochunnn Both boards are using same chip -> ESP32S3 with 16MB flash and 8MB OPI PSRAM. I have requested changes only for 1 board to match the chip specifications + also added some recommended partition tables.

Please do the changes for both boards if you agree :)

**Comment by Y1hsiaochunnn** [comment]
Hi, @P-R-O-C-H-Y According to the configuration parameters you mentioned, I have modified the configuration of two boards

**Review by Y1hsiaochunnn — COMMENTED** [review]
I have made the requested changes. Please review the updates.

**Review by P-R-O-C-H-Y — APPROVED** [review]
Looks good now :)

**Comment by P-R-O-C-H-Y** [comment]
@Y1hsiaochunnn When this PR gets merged, it would be nice if you also update the Waveshare wiki, that instead of selecting "S3 dev module" you can select the board itself? :)

**Comment by Y1hsiaochunnn** [comment]
Hi, @P-R-O-C-H-Y ，I will continue to monitor this PR merge status and update the wiki to select the boards as soon as possible after the merge

**Comment by Y1hsiaochunnn** [comment]
Hi, @P-R-O-C-H-Y ,Actually, what I don't quite understand is that if I use arduino-esp32 version 2.0.11, do some adaptation of the codebase and it works. When I upgrade to the latest version like 3.0.3, it doesn't work. Is this related to the custom library? Or is it related to the configuration of the updated arduino-esp32? Can you provide relevant documentation for reference?

**Comment by P-R-O-C-H-Y** [comment]
> Hi, @P-R-O-C-H-Y ,Actually, what I don't quite understand is that if I use arduino-esp32 version 2.0.11, do some adaptation of the codebase and it works. When I upgrade to the latest version like 3.0.3, it doesn't work. Is this related to the custom library? Or is it related to the configuration of the updated arduino-esp32? Can you provide relevant documentation for reference?

It's hard to answer as I don't properly understand the issue you are having. Do you have compilation issues or run-time issues? For 3.X we introduced some breaking changes to peripheral APIs so we also created a [Migration guide](https://docs.espressif.com/projects/arduino-esp32/en/latest/migration_guides/2.x_to_3.0.html#migration-from-2-x-to-3-0) to help with migrating projects/libraries from 2.X to 3.X version. Also this [Compatibility guide](https://docs.espressif.com/projects/arduino-esp32/en/latest/guides/core_compatibility.html#compatibility-guide-for-esp32-arduino-core) for libraries maintainers. It's explaining how your code base can support both 2.X and 3.X versions.

Feel free to open an issue with all informations, reproducible sketch and logs. So we don't spam this board addition as its not related :)

**Comment by Y1hsiaochunnn** [comment]
> > Hi, @P-R-O-C-H-Y ,Actually, what I don't quite understand is that if I use arduino-esp32 version 2.0.11, do some adaptation of the codebase and it works. When I upgrade to the latest version like 3.0.3, it doesn't work. Is this related to the custom library? Or is it related to the configuration of the updated arduino-esp32? Can you provide relevant documentation for reference?
> 
> It's hard to answer as I don't properly understand the issue you are having. Do you have compilation issues or run-time issues? For 3.X we introduced some breaking changes to peripheral APIs so we also created a [Migration guide](https://docs.espressif.com/projects/arduino-esp32/en/latest/migration_guides/2.x_to_3.0.html#migration-from-2-x-to-3-0) to help with migrating projects/libraries from 2.X to 3.X version. Also this [Compatibility guide](https://docs.espressif.com/projects/arduino-esp32/en/latest/guides/core_compatibility.html#compatibility-guide-for-esp32-arduino-core) for libraries maintainers. It's explaining how your code base can support both 2.X and 3.X versions.
> 
> Feel free to open an issue with all informations, reproducible sketch and logs. So we don't spam this board addition as its not related :)

Thank you for your answer. This is just a little confusion for me. I think I should follow what you said and check the Migration guide and Compatibility guide,If I have a new problem, I'll open a new issue

**Comment by Y1hsiaochunnn** [comment]
> @Y1hsiaochunnn - If you are able to update the Waveshare Wiki pages, could you also add that the Waveshare ESP32-S3-Matrix now has direct support starting with core v3.0.4 and can be selected from the Arduino IDE Boards manager.

Hi, @Xylopyrographer I wanna if when the v3.0.4 will be,If there is a message, I will plan it

**Pull request merged**
